project/ws/app/src/lib/routes/home/routes/request/competency-view/competency-view.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, Inject, OnInit } from '@angular/core'
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material'
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser'
+
+@Component({
+  selector: 'ws-app-competency-view',
+  templateUrl: './competency-view.component.html',
+  styleUrls: ['./competency-view.component.scss'],
+})
+export class CompetencyViewComponent implements OnInit {
+
+  levelSelected: any
+  constructor(
+    public dialogRef: MatDialogRef<CompetencyViewComponent>,
+    private sanitized: DomSanitizer,
+    @Inject(MAT_DIALOG_DATA) public dData: any
+  ) { }
+  ngOnInit() {
+    if (this.dData && this.dData.children && this.dData.children.length > 0) {
+      if (this.dData.selectedLevelId) {
+        this.levelSelected = this.dData.children.filter((v: any) => v.id === this.dData.selectedLevelId)[0]
+      } else {
+        this.levelSelected = this.dData.children[0]
```

Comment:
please validate children length, then assign 0

Reply:
already we are checking children length in line 19